### ISSUE TITLE
Allow for steps to accept a symbol

### ISSUE THREAD

**Issue by brennandunn** [elight/coulda#3]
We discussed this at DCamp: essentially, allow for something like:

Given :logged_in_as_admin
...

To work, and the symbol would fire a method of the same name. This would also be a no brainer for the pretty test runner output, "Given logged in as admin".

**Comment by elight** [comment]
As with Nick Gauthier's suggestion, also at Ruby DCamp, to allow Procs, in retrospect, I'm concerned for readability.

Your suggestion seems completely reasonable when considering 0-arity methods such as in your example.  However, how would n-arity methods be handled for n > 1?  It would probably be like one of the two following:

```
Given :logged_in_as_admin, admin_user_name, admin_user_password
```

or

```
Given admin_user_name, admin_user_password, :logged_in_as_admin
```

Neither of these is particularly readable.

This is where I would instead recommend doing the following:

```
def given_user_logged_in_as_admin(name, password)
  Given "A user logged in as admin" do
  .... 
  end
end
```

I'm uncomfortable treating 0-arity as a special case.

**Comment by brennandunn** [comment]
I see what you mean here, and you're right - looking at tests I've written, rarely do I have a helper method with no arity. i.e., #login_as(:admin). Plus, it only seems to make sense if you enforce that your helper methods accept an options hash:

  Given logged_in_as, :user => @admin

I think we're now on the same page here.

**Comment by brennandunn** [comment]
OK, too much to drink: I was trying to say that this would be the only viable option that makes sense:

Given :logged_in_as, :user => @admin

But that still becomes problematic when generating pretty output. So yeah, your example is ideal.

**Comment by elight** [comment]
Too much to drink already? ;-)

Closing the ticket.

**Comment by elight** [comment]
Reopening.  Actively considering this, with caution, for the next release.

**Comment by elight** [comment]
Implemented in Coulda 0.5.1.
